### PR TITLE
Add chart for HawkBit

### PR DESCRIPTION
**Pull request by axdotl**
Signed-off-by: Axel Köhler <axel.koehler@kiwigrid.com>

This PR will add the Hawkbit update server chart to the packages project.
Feedback is highly appreciated.

### CODE REVIEW

**Comment by thjaeckle** [comment]
Review required by @laverman as hawkBit representative in the Packages project

**Comment by monotek** [comment]
The problems seems to be usage of rabbitmq chart from bitnami repo, which is not added in ct.yaml

File: charts/hawkbit-update-server/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            - name: SPRING_PROFILES_ACTIVE
+              value: "{{ .Values.spring.profiles }}"
+            - name: MANAGEMENT_SERVER_PORT
```
[the remark this answers is not in the document]
Comment (axdotl) [reply charts/hawkbit-update-server/templates/deployment.yaml]: It configured the SpringBoot management server port. The port is used for the probes.
(ref: https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-features.html#production-ready-customizing-management-server-port)

File: charts/hawkbit-update-server/values.yaml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+spring:
+  profiles: "mysql"
+
+config:
+  application:
+    hawkbit:
+      ## Configuration for the device management federation
+      ## ref: https://www.eclipse.org/hawkbit/apis/dmf_api/
+      dmf:
+        hono:
```
[the remark this answers is not in the document]
Comment (axdotl) [reply charts/hawkbit-update-server/values.yaml]: I'm way unsure, but might you mean this PR eclipse/hawkbit#865?

**Comment by monotek** [comment]
Please don't merge. I want to try to fix the ci. Currently pvc were disabled.

**Comment by monotek** [comment]
CI works now with persistence by using a forked kind action: https://github.com/monotek/kind-action

I've create a pr at the helm folks kind action repo to get it merged: https://github.com/helm/kind-action/pull/16

Not sure if we want to go with my temporary version or wait for their merge.

**Comment by ctron** [comment]
Let's wait a day or two and see how it goes.

**Comment by ctron** [comment]
On another topic: When using the original helm charts for hawkbit from kiwigrid, I had to make some changes to be able to run with non-root containers. Is root still a requirement, or can we maybe get rid of that requirement from the start? I made a few modifications in order to drop the requirement, but I don't want to mess up your efforts :wink:

**Comment by monotek** [comment]
+1 for non root

As we're waiting anyway, we could maybe add the changes here?

**Comment by ctron** [comment]
Not to the changes, but to the content: https://github.com/ctron/hawkbit-kubernetes

**Comment by sophokles73** [comment]
@axdotl  @monotek This PR now seems to contain arbitrary changes to all kinds of artifacts, e.g. the general IoT Packages README, the CI pipeline and, last but not least, the hawkBit chart itself which originally seemed to be the subject of this PR :-)
I would appreciate if we could create separate PRs for the former two things as they do not seem to be related to the hawkBit chart per se, or are they?

**Comment by monotek** [comment]
Sure, no problem. I will create separate pr for the ci stuff.

**Comment by monotek** [comment]
i've created the pr: https://github.com/eclipse/packages/pull/23

**Comment by axdotl** [comment]
@sophokles73 With respect to the `charts/README.md`: @laverman asked for some info about a local setup. As this is similar to the setup described in the README of the Ditto chart, I assumed that this might be true for all charts in the package project. That's why I decided to provide some general info about a local setup.
What do you think: Should I move it to `charts/hawkbit/README.md` or should it remain in current location?

File: charts/hawkbit-update-server/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (laverman) [reply charts/hawkbit-update-server/values.yaml]: The properties in this section are not supported by hawkBit as of now. They will be introduced with  eclipse/hawkbit#890 as they configure Hono to be used as device registry for hawkBit via DMF. As these properties will have no effect at the moment, I was wondering whether to exclude them for now.

File: charts/hawkbit-update-server/templates/deployment.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (laverman) [reply charts/hawkbit-update-server/templates/deployment.yaml]: Understood. However, as there are no actual actuator endpoints (e.g. `health`, `metrics`, ...) available, I was wondering whether to just use the hawkBit landing page for the probes?

**Comment by ctron** [comment]
Could you re-base the PR?

File: charts/hawkbit-update-server/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (axdotl) [reply charts/hawkbit-update-server/values.yaml]: I commented the properties and added a reference to the related PR.
Is this okay for you?

File: charts/hawkbit-update-server/templates/deployment.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (axdotl) [reply charts/hawkbit-update-server/templates/deployment.yaml]: I changed it.

**Comment by ctron** [comment]
I added a small PR for this PR -> kiwigrid/packages#1 … This drops the default requirement to run as user 1001, and lets the cluster assign a random user uid, which makes it work on OpenShift. You can of course always re-enable the uid 1001 requirement.

**Comment by monotek** [comment]
I've added the needed stuff to run the contaienr as non root user now.

But as already mentioned in the comment to your pr (https://github.com/kiwigrid/packages/pull/1#pullrequestreview-347187701) i don't like to disable this by default.

**Comment by monotek** [comment]
> I still think the PR needs a rebase, but  in any case.

done.

**Comment by ctron** [comment]
Yay! Thanks for all the work!

**Pull request merged**